pkg/strutil/strutil_test.go

```diff
@@ -281,3 +281,11 @@ func TestParseCSVMap(t *testing.T) {
 		})
 	}
 }
+
+func TestOrderedQuery(t *testing.T) {
+	query := NewOrderedQuery()
+	query.Set("abc", "1")
+	query.Set("aa", "2")
+	result := query.Encode()
+	assert.Equal(t, result, "abc=1&aa=2", "")
+}
```

Comment:
Sorry, this is not correct, as the ordering is ignored on parsing back the string to `*net.URL`


https://github.com/containerd/containerd/blob/e85b5a0b81a610f564c3240c69d8f7324b5ecadb/pkg/process/io_util.go#L31

Reply:
May I ask if it is correct if we sort the args when in the nerdctl xmain function?

Reply:
> May I ask if it is correct if we sort the args when in the nerdctl xmain function?

No, that doesn't work with `nerdctl run busybox echo "_NERDCTL_INTERNAL_LOGGING"`


>  Maybe we don't need to concern about the args order?

Need